delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/model/FileEvent.scala

```diff
@@ -124,4 +144,51 @@ object FileEvent {
     */
   final case class FileDeprecated(id: Iri, project: ProjectRef, rev: Long, instant: Instant, subject: Subject)
       extends FileEvent
+
+  private val context = ContextValue(Vocabulary.contexts.metadata, contexts.files)
+
+  private val metadataKeys: Set[String] =
+    Set("subject", "types", "source", "project", "rev", "instant", "digest", "mediaType", "attributes", "bytes")
+
+  @nowarn("cat=unused")
+  implicit private val circeConfig: Configuration = Configuration.default
+    .withDiscriminator(keywords.tpe)
+    .copy(transformMemberNames = {
+      case "id"                                  => "fileId"
+      case "subject"                             => nxv.eventSubject.prefix
+      case field if metadataKeys.contains(field) => s"_$field"
+      case other                                 => other
+    })
+
+  @nowarn("cat=unused")
+  implicit def fileEventJsonLdEncoder(implicit
+      baseUri: BaseUri,
+      config: StorageTypeConfig
+  ): JsonLdEncoder[FileEvent] = {
+    implicit val subjectEncoder: Encoder[Subject] = Identity.subjectIdEncoder
+
+    implicit val encoder: Encoder.AsObject[FileEvent] = Encoder.encodeJsonObject.contramapObject { event =>
+      val storageAndType       = event match {
+        case created: FileCreated => Some(created.storage -> created.storageType)
+        case updated: FileUpdated => Some(updated.storage -> updated.storageType)
+        case _                    => None
+      }
+      implicit val storageType = storageAndType.map(_._2).getOrElse(StorageType.DiskStorage)
```

Comment:
Is there way to get rid of the default value ?

Reply:
Not really. But it doesn't matter much. This is just used here.